tests/pytorch/fused_attn/run_fused_attn_with_cp.py

```diff
@@ -45,6 +47,12 @@ def run_dpa_with_cp(dtype='bf16', model=None, qkv_format='bshd', kernel_backend=
 
     assert config.attn_mask_type in ['causal', 'no_mask'], f"{config.attn_mask_type} is an unsupported attention mask type!"
 
+    if kernel_backend == 'FusedAttention' and qkv_format == 'thd':
+        if 'causal' in config.attn_mask_type:
+            config.attn_mask_type = 'padding_causal'
+        else:
+            config.attn_mask_type = 'padding'
```

Comment:
Wouldn't `config.attn_mask_type` already be defined as `padding` or `padding_causal`? Just wondering what the purpose of this logic here is.

Reply:
In my test configs, I only have masking type of causal and no_mask [(refer here),](https://github.com/xrennvidia/TransformerEngine/blob/xren/cp_thd/tests/pytorch/fused_attn/test_fused_attn_with_cp.py#L48-L58) because padding with bshd and sbhd has not been supported with context parallelism yet. However, THD format is opposite, it only can work with padding or padding_causal, so I just simply change config.attn_mask_type like above to make THD test work.